packages/vite/src/node/plugins/css.ts

```diff
@@ -2277,6 +2298,80 @@ const makeModernScssWorker = (
   return worker
 }
 
+// this is mostly a copy&paste of makeModernScssWorker
+// however sharing code between two is hard because
+// makeModernScssWorker above needs function inlined for worker.
+const makeModernCompilerScssWorker = (
+  resolvers: CSSAtImportResolvers,
+  alias: Alias[],
+  _maxWorkers: number | undefined,
+) => {
+  let compiler: Sass.AsyncCompiler | undefined
+
+  const worker: Awaited<ReturnType<typeof makeModernScssWorker>> = {
+    async run(sassPath, data, options) {
+      // workaround for windows since import("D:...") fails
+      const sass: typeof Sass = createRequire(import.meta.url)(sassPath)
```

Comment:
I had `import(sassPath)` but it was failing on Windows due to drive `D:` prefix. Using `require` to just workaround this seems pretty odd. Does anyone know a safe way to normalize path into `file:...` for node esm?

Reply:
Maybe using `pathToFileURL`  from `node:url`:
`cachedSss = await import(sssPath.match(/^[a-z]:/i) ? pathToFileURL(sssPath).href : sssPath)`

Reply:
That might be it. I remembered Vitest had similar Windows hack somewhere and probably this one https://github.com/vitest-dev/vitest/blob/5d6d8013371b46522ff55cb64015d29e617994f2/packages/vite-node/src/client.ts#L368-L369
Linked issue on Node https://github.com/nodejs/node/issues/31710 seems to suggest `pathToFileURL(sssPath).href`.

Reply:
> btw, there is also `createRequire` for `sugarss` dependency

We should definitely use `import()` there, maybe in a separate PR. Good catch 👍